### PR TITLE
Add isolation segments

### PR DESCRIPTION
**Pull request by psycofdj**
This PR adds basic and experimental support for isolation segments. Relates to #8

Usefull links:
 - [v3 api specs](http://v3-apidocs.cloudfoundry.org/version/3.53.0/index.html#get-assigned-isolation-segment)

It also refines contributor documentation for running acceptance tests locally.

### CODE REVIEW

File: cloudfoundry/resource_cf_isolation_segment_test.go [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+						resource.TestCheckResourceAttr(entitleRef, "orgs.#", "2"),
+					),
+				},
+				resource.TestStep{
+					Config: segmentResourceUpdateOrgs,
+					Check: resource.ComposeTestCheckFunc(
+						testAccCheckSegmentExists(segRef, entitleRef),
+						resource.TestCheckResourceAttr(segRef, "name", "segment-one-name"),
+						resource.TestCheckResourceAttr(entitleRef, "orgs.#", "1"),
+					),
```
Comment (gberche-orange) [inline cloudfoundry/resource_cf_isolation_segment_test.go]: May be add coverage of resource import, similar to other imports, e.g. https://github.com/mevansam/terraform-provider-cf/blob/07283690098bbfa8fe5fbf4a8415ccdaba4b8933/cloudfoundry/import_cf_buildpack_test.go#L9

**Comment by gberche-orange** [comment]
Following exchange with @psycofdj, the pending limitations of this PR w.r.t. issue #8 (besides documentation refinements) were documented into https://github.com/mevansam/terraform-provider-cf/issues/8#issuecomment-410762321 for an upcoming distinct PR

**Comment by janosbinder** [comment]
LGTM - the tests failed due the missing mysql plan in acceptance environment - which has nothing to do with this PR. Please rebase or merge the large master in and address the comments.

**Comment by janosbinder** [comment]
@psycofdj could you please point exactly to the changes to contributions guidelines? I don’t find them.

**Comment by psycofdj** [comment]
@janosbinder I have trouble creating permalink on the markdown file
here is another link https://github.com/mevansam/terraform-provider-cf/pull/125/files#diff-6a3371457528722a734f3c51d9238c13

**Comment by janosbinder** [comment]
@psycofdj I have addressed some minor typos in the CONTRIBUTING.md . The local pcfdev took last time something like 40 mins to start.

File: cloudfoundry/repo/repo_manager.go [hunk 2]
```diff
@@ -107,6 +107,9 @@ func (rm *RepoManager) GetGitRepository(name string, repoURL string, user, passw
 
 // GetGithubRelease -
 func (rm *RepoManager) GetGithubRelease(ghOwner, ghRepoName, archiveName string, user *string, password *string) (repo Repository, err error) {
+	rm.gitMutex.Lock()
```
[the remark this answers is not in the document]
Comment (janosbinder) [reply cloudfoundry/repo/repo_manager.go]: Maybe we can create an issue and tackle this bug at a later point.

[the remark this answers is not in the document]
Comment (psycofdj) [reply cloudfoundry/repo/repo_manager.go]: In addition to opening a proper and separate issue;  would you mind merging this quick fix in this PR anyway  ?

**Comment by psycofdj** [comment]
@janosbinder thanks for the fixes in  CONTRIBUTING.md

File: cloudfoundry/repo/repo_manager.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (janosbinder) [reply cloudfoundry/repo/repo_manager.go]: No please go ahead. I think it is a very minor issue. We rarely use this functionality nor the buildpacks.

**Pull request merged**